### ISSUE TITLE
VCM not working, help debugging out

### ISSUE THREAD

**Issue by SiliconKite** [ackyshake/VimCompletesMe#74]
I use your plugin on two completely different Linux setups. On one it works with no setup required, in the other it does not work at all. I hope you can help me out or providing any ideas about what to look for?

Vim versions for NON working vim

```
VIM - Vi IMproved 8.2 (2019 Dec 12, compiled Mar 22 2021 00:00:00)
Included patches: 1-2637
Modified by <bugzilla@redhat.com>
Compiled by <bugzilla@redhat.com>
Huge version without GUI.  Features included (+) or not (-):
+acl               -farsi             +mouse_sgr         +tag_binary
+arabic            +file_in_path      -mouse_sysmouse    -tag_old_static
+autocmd           +find_in_path      +mouse_urxvt       -tag_any_white
+autochdir         +float             +mouse_xterm       -tcl
-autoservername    +folding           +multi_byte        +termguicolors
-balloon_eval      -footer            +multi_lang        +terminal
+balloon_eval_term +fork()            -mzscheme          +terminfo
-browse            +gettext           +netbeans_intg     +termresponse
++builtin_terms    -hangul_input      +num64             +textobjects
+byte_offset       +iconv             +packages          +textprop
+channel           +insert_expand     +path_extra        +timers
+cindent           +ipv6              +perl/dyn          +title
-clientserver      +job               +persistent_undo   -toolbar
-clipboard         +jumplist          +popupwin          +user_commands
+cmdline_compl     +keymap            +postscript        +vartabs
+cmdline_hist      +lambda            +printer           +vertsplit
+cmdline_info      +langmap           +profile           +virtualedit
+comments          +libcall           -python            +visual
+conceal           +linebreak         +python3/dyn       +visualextra
+cryptv            +lispindent        +quickfix          +viminfo
+cscope            +listcmds          +reltime           +vreplace
+cursorbind        +localmap          +rightleft         +wildignore
+cursorshape       +lua/dyn           +ruby/dyn          +wildmenu
+dialog_con        +menu              +scrollbind        +windows
+diff              +mksession         +signs             +writebackup
+digraphs          +modify_fname      +smartindent       -X11
-dnd               +mouse             -sound             -xfontset
-ebcdic            -mouseshape        +spell             -xim
+emacs_tags        +mouse_dec         +startuptime       -xpm
+eval              +mouse_gpm         +statusline        -xsmp
+ex_extra          -mouse_jsbterm     -sun_workshop      -xterm_clipboard
+extra_search      +mouse_netterm     +syntax            -xterm_save
   system vimrc file: "/etc/vimrc"
     user vimrc file: "$HOME/.vimrc"
 2nd user vimrc file: "~/.vim/vimrc"
      user exrc file: "$HOME/.exrc"
       defaults file: "$VIMRUNTIME/defaults.vim"
  fall-back for $VIM: "/etc"
 f-b for $VIMRUNTIME: "/usr/share/vim/vim82"
Compilation: gcc -c -I. -Iproto -DHAVE_CONFIG_H -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D_GNU_SOURCE -D_FILE_OFFSET_BITS=64 -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 
Linking: gcc -L. -Wl,-z,relro -Wl,-z,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -fstack-protector-strong -rdynamic -Wl,-export-dynamic -Wl,--enable-new-dtags -Wl,-z,relro -Wl,-z,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -Wl,-z,relro -Wl,-z,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -L/usr/local/lib -Wl,--as-needed -o vim -lm -lselinux -lncurses -lacl -lattr -lgpm -ldl -Wl,--enable-new-dtags -Wl,-z,relro -Wl,--as-needed -Wl,-z,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -Wl,-z,relro -Wl,--as-needed -Wl,-z,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -fstack-protector-strong -L/usr/local/lib -L/usr/lib64/perl5/CORE -lperl -lpthread -lresolv -ldl -lm -lcrypt -lutil -lc 
```

vim version for working plugin

```
VIM - Vi IMproved 8.2 (2019 Dec 12, compiled Jul 16 2020 16:33:06)
Included patches: 1-992
Compiled by root@
Huge version with GTK2 GUI.  Features included (+) or not (-):
+acl               -farsi             +mouse_sgr         +tag_binary
+arabic            +file_in_path      -mouse_sysmouse    -tag_old_static
+autocmd           +find_in_path      +mouse_urxvt       -tag_any_white
+autochdir         +float             +mouse_xterm       -tcl
-autoservername    +folding           +multi_byte        +termguicolors
+balloon_eval      -footer            +multi_lang        +terminal
+balloon_eval_term +fork()            -mzscheme          +terminfo
+browse            +gettext           +netbeans_intg     +termresponse
++builtin_terms    -hangul_input      +num64             +textobjects
+byte_offset       +iconv             +packages          +textprop
+channel           +insert_expand     +path_extra        +timers
+cindent           +ipv6              -perl              +title
+clientserver      +job               +persistent_undo   +toolbar
+clipboard         +jumplist          +popupwin          +user_commands
+cmdline_compl     +keymap            +postscript        +vartabs
+cmdline_hist      +lambda            +printer           +vertsplit
+cmdline_info      +langmap           +profile           +virtualedit
+comments          +libcall           -python            +visual
+conceal           +linebreak         +python3           +visualextra
+cryptv            +lispindent        +quickfix          +viminfo
+cscope            +listcmds          +reltime           +vreplace
+cursorbind        +localmap          +rightleft         +wildignore
+cursorshape       -lua               -ruby              +wildmenu
+dialog_con_gui    +menu              +scrollbind        +windows
+diff              +mksession         +signs             +writebackup
+digraphs          +modify_fname      +smartindent       +X11
+dnd               +mouse             -sound             -xfontset
-ebcdic            +mouseshape        +spell             +xim
+emacs_tags        +mouse_dec         +startuptime       +xpm
+eval              -mouse_gpm         +statusline        -xsmp
+ex_extra          -mouse_jsbterm     -sun_workshop      +xterm_clipboard
+extra_search      +mouse_netterm     +syntax            -xterm_save
   system vimrc file: "$VIM/vimrc"
     user vimrc file: "$HOME/.vimrc"
 2nd user vimrc file: "~/.vim/vimrc"
      user exrc file: "$HOME/.exrc"
  system gvimrc file: "$VIM/gvimrc"
    user gvimrc file: "$HOME/.gvimrc"
2nd user gvimrc file: "~/.vim/gvimrc"
       defaults file: "$VIMRUNTIME/defaults.vim"
    system menu file: "$VIMRUNTIME/menu.vim"
  fall-back for $VIM: "/usr/local/vim8/share/vim"
Compilation: gcc -std=gnu99 -c -I. -Iproto -DHAVE_CONFIG_H -DFEAT_GUI_GTK  -pthread -I/usr/include/gtk-2.0 -I/usr/lib64/gtk-2.0/include -I/usr/include/atk-1.0 -I/usr/include/cairo -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/pango-1.0 -I/usr/include/fribidi -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/harfbuzz -I/usr/include/freetype2 -I/usr/include/libpng15 -I/usr/include/uuid -I/usr/include/pixman-1 -I/usr/include/libdrm     -g -O2 -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1       
Linking: gcc -std=gnu99   -L/usr/local/lib -Wl,--as-needed -o vim   -lgtk-x11-2.0 -lgdk-x11-2.0 -latk-1.0 -lgio-2.0 -lpangoft2-1.0 -lpangocairo-1.0 -lgdk_pixbuf-2.0 -lcairo -lpango-1.0 -lfontconfig -lgobject-2.0 -lglib-2.0 -lfreetype   -lXpm -lXt -lX11 -lSM -lICE  -lm -ltinfo  -lselinux  -ldl     -L/usr/local/lib/python3.7/config-3.7m-x86_64-linux-gnu/ -lpython3.6m -lpthread -ldl -lutil -lm   
```

Other than slightly different versions, both config files look very similar. There is also some differences in the OS (not sure this matters) : one is Fedora 33 (using zsh bash and wayland if this matters), the other one is RedHat on tclsh shell

Any help very much appreaciated!

**Comment by ackyshake** [comment]
I'll need a little more information. 

Can you explain what you mean by "not working"? Does vimcompletesme.vim show up in the list of files when you invoke `:scriptnames`?

Where do you have VCM installed? Do you use a plugin manager?

**Comment by SiliconKite** [comment]
Sure, apologies for not being more explicit.

> Can you explain what you mean by "not working"? 
I mean it does autocomplete does not work for files on path, or for any omnicompletion on opened buffers.

> Does vimcompletesme.vim show up in the list of files when you invoke :scriptnames
Yes, it does.  `43: ~/.vim/bundle/VimCompletesMe/plugin/VimCompletesMe.vim`

> Where do you have VCM installed? Do you use a plugin manager?
I use vundle as plugin installer. Its installed on `/.vim/bundle/VimCompletesMe`.

Its weird because I have never had to do anything at all to make it work. It has always worked for me out of the box.

Thanks!

**Comment by SiliconKite** [comment]
Updated to describe what actually mean with does not work

**Comment by ackyshake** [comment]
You should see two files under `:scriptnames`, a plugin/VimCompletesMe.vim and an `autoload/VimCompletesMe.vim`.

It's possible that you have other plugins interfere with VCM's functionality. 

Can you do a simple test of running a vanilla vim (completely blank vimrc), and clone VimCompletesMe's repository into `~/.vim/pack/vendor/start`?

Then, please try opening Vim and trying to autocomplete something like this:

```
/usr/sha<hit tab>
```

**Comment by SiliconKite** [comment]
> You should see two files under :scriptnames, a plugin/VimCompletesMe.vim and an autoload/VimCompletesMe.vim.

Hmmm, I just see one, the one I mentioned before... I guess this is the issue then? Maybe I can force the autoload?

> Can you do a simple test of running a vanilla vim (completely blank vimrc), and clone VimCompletesMe's repository into

I cloned VCM into `$HOME/.vim/pack/vendor/start/` so path is now `$HOME/.vim/pack/vendor/start/VimCompletesMe`. I invoke an empty vim with `vim -u none`. Autocompletion does not work unfortunately

**Comment by ackyshake** [comment]
> > You should see two files under :scriptnames, a plugin/VimCompletesMe.vim and an autoload/VimCompletesMe.vim.
> 
> Hmmm, I just see one, the one I mentioned before... I guess this is the issue then? Maybe I can force the autoload?

You shouldn't need to do so manually. It should be autoloaded when you first try to trigger a completion with the Tab key.

> > Can you do a simple test of running a vanilla vim (completely blank vimrc), and clone VimCompletesMe's repository into
> 
> I cloned VCM into `$HOME/.vim/pack/vendor/start/` so path is now `$HOME/.vim/pack/vendor/start/VimCompletesMe`. I invoke an empty vim with `vim -u none`. Autocompletion does not work unfortunately

vim -u NONE will not stop loading plugins that are already in the runtimepath. Please try to either move any plugins out of the way by packing up your old ~/.vimrc or try using --noplugin as well (See `:help --noplugin`). It's possible Fedora/your vim package installs something specific to that distribution.

Anyway, just installing VCM and then opening up Vim without any other plugins and a blank vimrc file, then running running `:scriptnames` should list the plugin file fo VCM. Pressing the tab key after trying to complete should then list the autoload file for VCM as well (after running `:scriptnames` again)

**Comment by SiliconKite** [comment]
I am not sure how this issue fixed by itself. I installed a vim with GUI support and then it started working, even in the vim version that has no GUI suppot. I guess the vim I had installed (via Fedora repo) might have been missing some lib needed by VCM?

Anyway, for anybody else getting into this issue, installing vim-X11 package on Fedora (might have different name on other distros) fixed the issue.

Thanks so much for the support
